hack/operatorhub/internal/operatorhub/operatorhub.go

```diff
@@ -0,0 +1,650 @@
+		DigestPinning       bool   `json:"digestPinning"`
+	} `json:"packages"`
+}
+
+// hasDigestPinning will return true if any package
+// within the configuration has DigestPinning enabled.
+func (c *config) hasDigestPinning() bool {
+	if c == nil {
+		return false
+	}
+	for _, pkg := range c.Packages {
+		if pkg.DigestPinning {
+			return true
+		}
+	}
+	return false
+}
+
+func loadConfigFile(c GenerateConfig) (*config, error) {
+	confBytes, err := ioutil.ReadFile(c.ConfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("while opening file %s: %w", c.ConfigPath, err)
+	}
+
+	var conf config
+	if err := gyaml.Unmarshal(confBytes, &conf); err != nil {
+		return nil, fmt.Errorf("while unmarshaling config from %s: %w", c.ConfigPath, err)
+	}
+
+	// allow command line flags to override config file
+	if c.NewVersion != "" {
+		conf.NewVersion = c.NewVersion
+	}
+	if c.PrevVersion != "" {
+		conf.PrevVersion = c.PrevVersion
+	}
+	if c.StackVersion != "" {
+		conf.StackVersion = c.StackVersion
+	}
+
+	return &conf, nil
+}
+
+func getInstallManifestStream(conf *config, manifestPaths []string) (io.Reader, func(), error) {
+	if len(manifestPaths) == 0 {
+		s, err := installManifestFromWeb(conf.NewVersion)
+		return s, func() {}, err
+	}
+
+	var rs []io.Reader
+	closer := func() {
+		for _, r := range rs {
+			if closer, ok := r.(io.Closer); ok {
+				closer.Close()
+			}
+		}
+	}
+	for _, p := range manifestPaths {
+		r, err := os.Open(p)
+		if err != nil {
+			return nil, closer, fmt.Errorf("while opening %s: %w", manifestPaths, err)
+		}
+		rs = append(rs, r)
+		// if we're using local yaml files, ensure that they have a proper
+		// end of directives marker between them.
+		rs = append(rs, strings.NewReader(yamlSeparator))
+	}
+	return io.MultiReader(rs...), closer, nil
+}
+
+func installManifestFromWeb(version string) (io.Reader, error) {
+	// try the legacy all-in-one first for older releases
+	buf, err := makeRequest(fmt.Sprintf(allInOneURL, version))
+	if err == errNotFound {
+		// if not found load the separate manifests for CRDs and operator (version >= 1.7.0)
+		crds, err := makeRequest(fmt.Sprintf(crdManifestURL, version))
+		if err != nil {
+			return nil, err
+		}
+		op, err := makeRequest(fmt.Sprintf(operatorManifestURL, version))
+		if err != nil {
+			return nil, err
+		}
+		return io.MultiReader(crds, strings.NewReader(yamlSeparator), op), nil
+	}
+	return buf, err
+}
+
+func makeRequest(url string) (io.Reader, error) {
+	client := http.Client{}
+
+	ctx, cancelFunc := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancelFunc()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, http.NoBody)
+	if err != nil {
+		return nil, fmt.Errorf("while creating request to %s: %w", url, err)
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("while executing HTTP GET %s: %w", url, err)
+	}
+
+	defer func() {
+		if resp.Body != nil {
+			resp.Body.Close()
+		}
+	}()
```

Comment:
https://github.com/elastic/cloud-on-k8s/commit/1168b13c6606034d1d1f43fcce207918df2d2ce0

```suggestion
	defer resp.Body.Close()
```